src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java

```diff
@@ -119,19 +113,7 @@ public FileUploadSubmission handleFileUploadSubmission(FileUploadSubmission file
         // We take all the results in this exercise associated to the tutor, and from there we retrieve the submissions
         List<Result> results = this.resultRepository.findAllByParticipationExerciseIdAndAssessorId(exerciseId, tutorId);
 
-        return results.stream().map(result -> {
-            Submission submission = result.getSubmission();
-            FileUploadSubmission fileUploadSubmission = new FileUploadSubmission();
-
-            result.setSubmission(null);
-            fileUploadSubmission.setLanguage(submission.getLanguage());
-            fileUploadSubmission.setResult(result);
-            fileUploadSubmission.setParticipation(submission.getParticipation());
-            fileUploadSubmission.setId(submission.getId());
-            fileUploadSubmission.setSubmissionDate(submission.getSubmissionDate());
-
-            return fileUploadSubmission;
-        }).collect(Collectors.toList());
+        return results.stream().map(result -> mapAbstractToConcreteSubmission(result, new FileUploadSubmission())).collect(Collectors.toList());
```

Comment:
Why do we need to clone the submission object? `Submission` is an abstract class, you should get a list of `FileUploadSubmissions` anyway.

Reply:
In this case we are doing a deep copy of submission. So we need to have a new instance of object, so it should be concrete.

Reply:
But we still don't need to do a deep copy here. This method should only return a list of submissions without any modifications. You could either just unproxy the submission objects and or use a different query altogether (which I think is even better).
E.g. this query for `FileUploadSubmissions`:
```suggestion
        return submissionRepository.findAllByResult_Participation_ExerciseIdAndResult_Assessor_Id(exerciseId, tutorId).stream()
            .map(submission -> (FileUploadSubmission) submission)
            .collect(Collectors.toList());
```

Reply:
@ungaralex Thanks for pointing out the solution with the query. It works like a charm